### PR TITLE
add H2 support

### LINKED ISSUE #7: add H2 support
This feature needs implementation of [H2 full text search](http://h2database.com/html/tutorial.html#fulltext). It would allow the app to run using an embedded H2 db without relying on PostgreSQL.

### PR DESCRIPTION
**Pull request by jhamrsky**
This PR adds H2 DB support and closes #7. 

Roughly, following was done:
* Unified SQL syntax by removing "RETURNING" statement, which is specific for PG.
  * Used HugSQL feature of `.getGeneratedKeys` in combination with custom method.
  * Method names were left with YeSQL syntax using `<!` in order to somehow mark that the caller should keep in mind returning value.
  * Introduced `*db-type*` based on JDBC connection string which can be used for SQL-dialect specific purposes.
* Added H2 support and files
  * Migrations are based on PG migrations. Most changes just remove PG specific calls.
  * For SQL compabibility of calls of `array_agg`, `ArrayAggregator` was added to code (Java).
  * Extended interface on required classes for reading parameters.
  * Adjusted project/main file to support different migration schema. PG set as default one.
* Added some text to readme about H2.
* Tested functionalities with PG and H2.

If it comes to full text search, H2 support is somewhat limited (no weighting support). I intentionally left out wildcard support (has some performance issues) and used native search engine. But otherwise, H2 is nice for more lightweight use cases.

### CODE REVIEW

**Comment by yogthos** [comment]
Hi,

Thanks for the PR, this looks pretty comprehensive. :)

File: README.md [hunk 1]
```diff
@@ -83,6 +83,8 @@ Create a `profiles.clj` file in the project directory with the configuration set
 ```clojure
 {:profiles/dev
  {:env
+  ;; set migratus dir (and :migrations-dir) when developing with H2
```
Comment (yogthos) [inline README.md]: I think this needs to be expanded a bit. I'd not in the section above that you can either use H2 or Postgres, and give separate config examples for both cases.

Comment (jhamrsky) [reply README.md]: You are right, I guess I was a little bit too much brief about the usage. I was thinking about how it should be described and I see two options. 
1. For each DB type full individual sample differentiated only by the database setup. 
1. Individual smaller parts describing related part of the `profiles.clj` (currently three: PGSQL, H2, LDAP). 

Both have its ups and downs. 
1. Is little bit more chatty and worse for maintance (eg. LDAP gets something new), but copy paste friendly. 
1. Is easier for maitanance and easily extensible (eg. new supported DB), but requires some kind of ellipsis and forces developer to glue related parts together.

What would be your preference regarding this?

Comment (yogthos) [reply README.md]: From playing around with it locally, it looks like the only thing you have to change is the db URL with the new setup.  So, I'd opt for just changing the "Configuring PostgreSQL" database section into "Configuring the Database", and adding subsections for H2 and Postgres under it.

**Pull request merged**